cupyx/scipy/sparse/csr.py

```diff
@@ -477,6 +486,15 @@ def isspmatrix_csr(x):
     return isinstance(x, csr_matrix)
 
 
+def check_shape_for_pointwise_op(a_shape, b_shape):
```

Comment:
I noticed SciPy doesn't check the input shapes. Can't we just let the division raise the error if the shapes are incompatible?

Reply:
(I do see this was implemented elsewhere and this is just a refactoring, so I'm simply wondering and not requesting a change 😅)

Reply:
The input shapre is checked here.
https://github.com/scipy/scipy/blob/3af12341854f9f9a6ef0a3a9c853e80bcc5e8002/scipy/sparse/compressed.py#L1249-L1250
The difference is that this PR allows broadcasting (like point-wise multiply), while the SciPy does not..

Reply:
OK, I will fix it to raise an exception when broadcasting is needed for compatibility with SciPy.